### DISCUSSION TITLE
Announcing v2.2.0 - what's new

### DISCUSSION THREAD

**Discussion by jw-foss** [element-plus/element-plus#7578] [Announcement]
## Features (新特性) 🚀 

For `2.2.0`, we finally brought **dark mode** and **Virtualized Table** into Element Plus.
These are the biggest new features we are making this `minor` version about.

- For enabling dark mode, you might want to see [Enable dark mode](https://element-plus.org/en-US/guide/dark-mode.html)
- For how to use Virtualized Table, please refer to [Documentation for Virtualized Table](https://element-plus.org/en-US/component/table-v2.html)

If you find any issues while using these new features, please do not hesitate to report a bug.

### Virtualized Table（虚拟化表格）

Virtualized Table is just like the other two features we released before `Virtualized Select` and `Virtualized Tree`.
虚拟化表格就跟它的两个姊妹组件 -- `虚拟化选择器` 和 `虚拟化树` 一样，带来更高性能的表现。

We had a lot of trade-offs on this component. Which API needs to be provided, and which does not. For the existing table component, a lot of APIs were given, making the component so complicated and difficult be maintained. So this time, we tried to provide a minimal set of APIs without sacrificing the ability of customization.

在做这个组件的过程中，我们遇到了很多需要权衡的地方。哪些 API 我们应该提供，哪些我们不应该。在现有的 Table 组件里，我们提供了大量的 API，这导致这个组件极其复杂并且难以维护。所以这一次我们尝试在提供那些必须的 API 的同时不牺牲可定制性。

This comes with a price which users are now in response to do the job themselves. As I said, there were trade-offs, for some users who don't need advanced features it would be a good thing, because they don't need to include all that unused code in their apps. For advanced users, this could be bad news because they will have to implement the features provided by table-v1 themselves, but we made a bunch of examples for helping you, basically, everything you need is included in these examples.
If you find that something was missing in the examples, you can make a PR to us or report an issue.

当然，这样做是有一定代价的 -- 现在用户需要自己来实现一些他们想要的功能。这些就像我上面提到的权衡点：
- 对一部分用户来说，他们可能不太需要那么高级的功能，但是这这部分功能的代码却内置在组件中，也会在他们的项目里面加入没有用的代码。
- 对另一部分用户来说，有一些 Table V1 提供的功能没有通过 API 的方式提供出来，会是不太方便的事情，对于这点我们也在努力尝试能尽量解决这些担忧，我们在文档的用例代码里添加了很多丰富的功能实现方式，你只需要依葫芦画瓢就能很快实现你想要的各种功能功能。如果你发现有一些内容是你没有在文档里看到的话，我们非常欢迎你来提一个 issue 或者是直接提交一个 Pull Request。

There could be bugs based on the way you use it. But if you find any you can also report to us or make a PR.

这个组件可能会有潜在的 bug，如果你觉得你发现了一个 bug，欢迎向我们提交 issue 或者 Pull Request。

### A11y features（可访问性）

Shoutout to @Giwayume who made Element Plus more accessible for users with screen readers. Now we have a11y features:

在 @Giwayume 的大力帮助下，Element Plus 现在对屏幕阅读器（如 VoiceOver, Microsoft Narrator）

1. Form items will automatically attach labels to inputs #7450 (Form Items 现在会自动给它的后辈元素添加 Label 属性)
2. Slider component has `aria` attributes #7389 （Slider 组件现在有了 ARIA 属性，能够更好地使用键盘来操作）

### Dark mode（暗黑模式）

With @YunYouJun and @huangzhongzhong's hard-working, we extracted most of our design tokens and implemented this feature on top of that. 

在 @YunYouJun 和 @huangzhongzhong 的辛勤努力下，我们把很多设计的 Token 整理了出来，并在这个基础之上实现了暗黑模式。

### Text Button（文字类的按钮）

We removed `type="text"` from Button component use an equivalent API **text** for doing that. We also made visual changes to `Text Button`。https://github.com/element-plus/element-plus/pull/7515

我们移除掉了 Button 的 `type="text"` 这个选项，新加入了一个 **text** 字段来启用文字类的按钮。我们也对文字类的按钮做了视觉效果的更新。详情见 https://github.com/element-plus/element-plus/pull/7515

Before this change, `Text Button` did not really look like a button, more like a text, so we made a visual enhancement for it to make it more `Button`. And because `type="xx"` also represents the color of each button, we decided to introduce a new API for `Text Button`

在这个改动之前，文字类的按钮看起来不太像一个按钮反而比较像一段文字，所以我们对它进行了视觉效果的升级让它看起来更像是按钮而不是文字。由于 `type="xx"` 同时决定了对应按钮的颜色属性，所以我们决定加入一个新的 API 来区分文字类的按钮和其它类型的按钮。

## Furthermore（接下来的一些计划）

For the next upcoming minor version, we are planning to release the missed out  `Typography` component which allows you easily put your texts to the Apps. 

在接下来的一个 minor 版本中，我们计划加入 `Typography` 组件来方便快速地文字排版。

We are also working on the a11y features with @Giwayume. 

我们也在和 @Giwayume 一起努力为 Element Plus 添加更多可访问性的功能。

We are also planning on making a **Theme Market** so you can define your own theme easily. 

我们同时也在计划发布一个**主题市场**的应用，方便你可以自定义你独有的 Element Plus 主题。

## Appreciations（特别感谢）

Many thanks to our fellow developers, sponsors, and users.  Thank you all for being so supportive, Element Plus is here because of every one of you:heart: We will continue to excel in our works, making Element Plus better!

非常感谢我们的开发者，赞助商，以及用户们。正因为你们的支持与包容，我们才能走到这里:heart: 我们会继续努力把 Element Plus 做得更好

## Get involved（参与进来）

If you are interested in making Element Plus better, you can join our [Discord](https://discord.link/ElementPlus) to get in touch with the team, and the community.

如果你对 Element Plus 的开发或建议感兴趣，你可以加入我们的 [Discord](https://discord.link/ElementPlus) 频道来和我们快速取得联系，与社区进行沟通。

**Comment by jw-foss** [comment]
感谢大家的支持，我们会继续加油的，由于个人精力有限，有时候真的没办法回复那么快和全，实在抱歉 :bow:

**Reply by ishowman** [reply]
> 感谢大家的支持，我们会继续加油的，由于个人精力有限，有时候真的没办法回复那么快和全，实在抱歉 🙇

理解，我觉得更好的选择是降低 element-plus 的开发上手成本。我想贡献代码，但看现有文档，我实在不明白组件源码更改后，要如何自动生成组件类型代码。希望能完善这方面的信息，降低想贡献代码的学习成本   @jw-foss 能反馈下这个问题么？

**Reply by YunYouJun** [reply]
类型一般是 dts 编译所得，执行 `vue-tsc --noEmit` 可进行编译。

**Comment by zouhangwithsweet** [comment]
Great job! 💯

**Comment by iamkun** [comment]
amazing 😘

**Comment by kooriookami** [comment]
Awesome!

**Comment by banbri** [comment]
Good job!

**Comment by snvtac** [comment]
Awesome

**Comment by satrong** [comment]
希望能在 release 页面中加一项 breaking 列表

**Reply by jw-foss** [reply]
@satrong https://github.com/element-plus/element-plus/releases/tag/2.2.0 Added. 👍

**Reply by satrong** [reply]
@JeremyWuuuuu  Nice!

**Comment by wkl007** [comment]
期待主题市场

**Comment by bartlian** [comment]
well done

**Comment by eybarta** [comment]
Hey, thanks for you awesome work!

I've been looking for a data table component to be used in our product,
and after going through a bunch I finally found your library. 
I really love the simplicity and your great documentation.

We definitely need the virtual table because we deal with large amounts of data and performance is a requirement.

Looking at this thread, it's been about a year to the day since the release.. I've been looking over some 
of the issues and open PR's.. and see some of them hanging for months with no reply or review from maintainers.

So my main concern is how still actively maintained is this library and feature. Is there a roadmap to have the virtual-table
come out of 'beta' and be production ready?

Can we trust that the team at element-plus will be responsive and review any comments/issues/PR's we open?

I appreciate your time and hopeful to get a response on the status of development so we can make a decision :)

Thank you.

**Comment by webvs2** [comment]
`Typography `  Is it already under development? If possible, I would like to develop it as an alternative

**Reply by YunYouJun** [reply]
What are the features of typography you are looking forward to?

**Comment by Jackie-llv** [comment]
wow
